crates/net/network/src/transactions/mod.rs

```diff
@@ -598,6 +601,7 @@ where
         &mut self,
         peer_id: PeerId,
         mut msg: NewPooledTransactionHashes,
+        cx: &mut Context<'_>,
```

Comment:
this part feels a bit weird to me, because what this does is buffer more responses and it's not very clear to me what the benefit here is.

This only has the advantage to maybe complete an inflight request to the peer that sent the announcement so we can send a request right away?

Though, I think this can also be achieved by advancing/polling the fetcher before handling new incoming commands, right? because that's what `on_new_pooled_transaction_hashes` does anyway

This also means that with this change, we're only advancing in progress requests when we receive new incoming hashes or trying to create new outgoing requests (on_fetch_pending_hashes)

we can get the same effect by changing the loop order, by advancing in progress requests first.

freeing up request capacity on new incoming hashes so we can immediately send a request, might also not be beneficial because there's a chance that we receive the tx in full.

Reply:
advancing inflight requests on-op is better, since between the point in time that tx fetcher is polled for fetch events and the point when processing of an announcement leads to queueing more hashes for fetch, some inflight requests may have mad progress. as stated in pr description, the limit is 130 available peers. if using the same default max size of expected response as geth, 128 KiB, a request is filled with tops 1,3k legacy txns of 100 bytes each. an announcement can have 4k hashes. that's 3 requests that can be assembled from one announcement. I think it is obvious 3:1 that lack of idle peer can become bottleneck.

as to the part about using this bottleneck to avoid fetching hashes that might be received over broadcast, allow me to make an analogy. would you consider it better to have an old malfunctioning car to ensure you don't to drive too fast, or to have a Tesla and learn how to drive it slow? surely we would place an intentional mechanism to delay requests if we can prove that useful with benchmarks.